TypeScript failure

The [last release](https://github.com/nodejs/undici/releases/tag/v4.14.0) introduced a TypeScript bug that does not allow TypeScript to compile. It looks like the bug has been introduced in https://github.com/nodejs/undici/pull/1200.

Error:
```
node_modules/undici/types/mock-interceptor.d.ts:74:14 - error TS2304: Cannot find name 'Headers'.

74     headers: Headers;
                ~~~~~~~
```

To reproduce, create a empty project, install `undici`, `typescript` and `@types/node` and try to compile the following:
```ts
import { Request } from 'undici'
```

I don't know how to fix it, if anyone has an idea I can open a pr.

--- a/package.json
+++ b/package.json
@@ -109,7 +109,8 @@
   "tsd": {
     "directory": "test/types",
     "compilerOptions": {
-      "esModuleInterop": true
+      "esModuleInterop": true,
+      "lib": ["esnext"]
     }
   }
 }

--- a/test/types/fetch.test-d.ts
+++ b/test/types/fetch.test-d.ts
@@ -17,6 +17,7 @@ import {
   Response,
   ResponseInit,
   ResponseType,
+  ReferrerPolicy
 } from '../..'
 
 const requestInit: RequestInit = {}

--- a/test/types/proxy-agent.test-d.ts
+++ b/test/types/proxy-agent.test-d.ts
@@ -1,4 +1,5 @@
 import { expectAssignable } from 'tsd'
+import { URL } from 'url'
 import { ProxyAgent, setGlobalDispatcher, getGlobalDispatcher, Agent } from '../..'
 
 expectAssignable<ProxyAgent>(new ProxyAgent(''))

--- a/types/mock-interceptor.d.ts
+++ b/types/mock-interceptor.d.ts
@@ -1,5 +1,6 @@
 import { IncomingHttpHeaders } from 'http'
 import Dispatcher from './dispatcher';
+import { Headers } from './fetch'
 
 export {
   Interceptable,

